src/Builder/Encoder/PipelineEncoder.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Builder\Encoder;
+
+use MongoDB\Builder\Pipeline;
+use MongoDB\Codec\EncodeIfSupported;
+use MongoDB\Exception\UnsupportedValueException;
+use stdClass;
+
+/** @template-extends AbstractExpressionEncoder<array, Pipeline> */
+class PipelineEncoder extends AbstractExpressionEncoder
+{
+    /** @template-use EncodeIfSupported<array, Pipeline> */
+    use EncodeIfSupported;
+
+    public function canEncode(mixed $value): bool
+    {
+        return $value instanceof Pipeline;
+    }
+
+    public function encode(mixed $value): stdClass|array|string
```

Comment:
This returns always a list.

```suggestion
    public function encode(mixed $value): array
```

Reply:
In this case, would `list` be preferable?

Reply:
I changed the return type to `array`, and the resulting type to `list<array|stdClass|string>` as `list` itself wasn't sufficient. Since we don't have `BuilderEncoder::encode` typed any more strictly, this is the best we can do.

Reply:
Correction: it's `list<mixed>`, as `encodeIfSupported` is used throughout the encoders, which results in an effective return type of `mixed` whenever we defer to the `BuilderEncoder` class.